### DISCUSSION TITLE
Addon and Scratch Desktop

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#1724] [Discussion]
Is there a way, after installing the offline Scratch Desktop, to add the addons to this app?

**Comment by scratchusernamemrtbts** [comment]
that would be cool

**Comment by scratchusernamemrtbts** [comment]
but I think that should be made an app instead of a browser extension

**Comment by mxmou** [comment]
[TurboWarp Desktop](https://desktop.turbowarp.org/) is a Scratch Desktop modification that includes some features from Scratch Addons.

**Reply by sfederici** [reply]
> [TurboWarp Desktop](https://desktop.turbowarp.org/) is a Scratch Desktop modification that includes some features from Scratch Addons.

I didn't notice the Desktop version of TurboWarp. But I would like to be able to select all the interesting addons of Scratch Addons :-)

**Reply by WorldLanguages** [reply]
@sfederici TurboWarp has pretty much all of our editor addons

**Reply by sfederici** [reply]
I know, but they are not up to date for what concerns fixes. Two of the addons I use most still have old glitches that have been removed weeks from Scratch Addons.

**Reply by jeffalo** [reply]
turbowarp follows a different release cycle, but it will have the latest addons when it gets updated. im not sure how it works, but i think garbomuffin has a script that automatically builds the addons from from this repo into turbowarp.

cc @GarboMuffin

**Reply by GarboMuffin** [reply]
Addons are updated or added added each stable release within a couple days. Bug fixes sometimes get cherry-picked early.

TW Desktop is updated much less frequently than the website, the last update was nearly 3 weeks ago. The next update will be early this week, Wednesday at the latest. Moving forward it's going to be updating more often.

**Comment by Hans5958** [comment]
Other than TW Desktop, is it possible to make a mod of Scratch, just like BetterDiscord?

**Reply by WorldLanguages** [reply]
Technically we could?
Note: actually, depends on what you mean by mod. We'd fork the editor and need to run our own update server and such

**Reply by Mr-Programmer-777** [reply]
> Technically we could?
> Note: actually, depends on what you mean by mod. We'd fork the editor and need to run our own update server and such

I think that would be awesome, ScratchWarp or something...
Maybe if you did that, you could add a JS, Python, or HTML code input?
So, you have ScratchBlocks, and, a script box for text code? I think that could be an amazing thing to have; it would let you change the code in real-time, from static to being able to change,,,

**Reply by Hans5958** [reply]
> Note: actually, depends on what you mean by mod. We'd fork the editor and need to run our own update server and such

No. What I mean is to have a mechanism so we can inject our scripts to the Scratch app, just like BetterDiscord inject some scripts to run. It's an Electron app after all.
